lib/fizzy/execute.cpp

```diff
@@ -315,7 +315,12 @@ inline T load(bytes_view input, size_t offset) noexcept
 template <typename DstT, typename SrcT>
 inline DstT extend(SrcT in) noexcept
 {
-    if constexpr (std::is_signed<SrcT>::value)
+    if constexpr (std::is_floating_point_v<SrcT>)
```

Comment:
I don't think I like additional levels of generic helpers. But in this case I propose simpler implementation.
```
if constexpr (std::is_same_v<SrcT, DstT>)
    return in;

// Handle else...
```